slurm/slurm.md

```diff
@@ -0,0 +1,941 @@
+```
+
+Start the slurmdbd service
+
+```
+systemctl enable slurmdbd
+systemctl start slurmdbd
+```
+
+### On the `master.local` machine
+
+
+We will be creating the `/etc/slurm/slurm.conf` file, that comprises of all the configuration that slurm needs in order to function properly, more information about `slurm.conf` can be found [here](https://slurm.schedmd.com/slurm.conf.html).
+
+```
+#General Cluster Configurations
+ClusterName=production
+SlurmctldHost=master(X.X.X.X)
+SlurmctldHost=shadow(X.X.X.X)
+
+# Job Management
+DisableRootJobs=NO
+EnforcePartLimits=NO
+GroupUpdateForce=1
+GroupUpdateTime=600
+JobFileAppend=0
+JobRequeue=1
+KillOnBadExit=1
+LaunchType=launch/slurm
+MailProg= /usr/bin/smail
+MaxJobCount=5000
+MaxStepCount=40000
+MaxTasksPerNode=128
+MpiDefault=pmi2
+ProctrackType=proctrack/cgroup
+PropagatePrioProcess=0
+#JobSubmitPlugins=lua
+PropagateResourceLimits=MEMLOCK,STACK
+ReturnToService=0
+SlurmctldPidFile=/var/run/slurmctld.pid
+SlurmctldPort=6817
+SlurmdPidFile=/var/run/slurmd.pid
+SlurmdPort=6818
+SlurmdSpoolDir=/var/spool/slurmd
+SlurmUser=slurm
+StateSaveLocation=/var/spool/slurm/job_checkpoint
+SwitchType=switch/none
+TaskPlugin=task/affinity,task/cgroup
+TaskPluginParam=NONE
+RebootProgram=/usr/sbin/reboot
+TmpFS=/tmp
+TrackWCKey=no
+#TreeWidth=4
+UsePam = 1
+#
+
+
+# Timers
+BatchStartTimeout=10
+CompleteWait=3
+InactiveLimit=600
+KillWait=30
+MessageTimeout=10
+ResvOverRun=0
+MinJobAge=300
+OverTimeLimit=0
+SlurmctldTimeout=120
+SlurmdTimeout=300
+UnkillableStepTimeout=60
+VSizeFactor=0
+Waittime=0
+#
+
+#Job Scheduling
+SchedulerTimeSlice=30
+SchedulerType=sched/backfill
+SchedulerParameters=nohold_on_prolog_fail
+SelectType=select/cons_res
+SelectTypeParameters=CR_Core
+
+# Job Priority
+PriorityType=priority/multifactor
+PriorityFlags=FAIR_TREE
+## 2 week half-life
+PriorityDecayHalfLife=14-0
+# The larger the job, the greater its job size priority.
+PriorityFavorSmall=NO
+# The job's age factor reaches 1.0 after waiting in the
+# queue for 2 weeks.
+PriorityMaxAge=14-0
+# This next group determines the weighting of each of the
+# components of the Multi-factor Job Priority Plugin.
+# The default value for each of the following is 1.
+PriorityWeightAge=1000
+PriorityWeightJobSize=1000
+PriorityWeightPartition=10000000
+PriorityWeightQOS=10000000
+PriorityWeightTRES=CPU=2000,Mem=4000
+PriorityWeightFairshare=100000
+#
+
+# LOGGING AND ACCOUNTING
+AccountingStorageType=accounting_storage/slurmdbd
+AccountingStorageHost=X.X.X.X
+#AccountingStoreFlags=job_comment
+AccountingStoragePort=6819
+AccountingStorageEnforce=limits,associations,qos
+AccountingStorageTRES=CPU,Mem
+SlurmctldLogFile=/var/log/slurmctld.log
+SlurmdLogFile=/var/log/slurmd.log
+JobAcctGatherFrequency=30
+JobAcctGatherType=jobacct_gather/cgroup
+SlurmctldDebug=info
+SlurmdDebug=info
+SlurmSchedLogFile=/var/log/slurmSched.log
+SlurmSchedLogLevel=1
+#
+#Energy consumption test
+AcctGatherEnergyType=acct_gather_energy/rapl
```

Comment:
Maybe add comment that this can be very wrong for some newer Intel CPUs. Check: https://bugs.schedmd.com/show_bug.cgi?id=9956

Reply:
We removed it altogether, since is was just a test.